### ISSUE TITLE
encodeImage no support for compression

### ISSUE THREAD

**Issue by varun-raghavendra** [photopea/UTIF.js#121]
UTIF.encodeImage currently encodes a new Tiff with no compression. This blows up the size of the output TIFF. 

Is there a plan to support different compressions like LZW, G4, etc for encodeImage? 

Also, could you provide any pointers to build my own image encoder with compression? Unsure how to start off with this.

**Comment by varun-raghavendra** [comment]
@jaddi, You've mentioned in the below issue that it shouldn't be hard to use existing compression while encoding new Tiff image. Could you elaborate?

[@jaddi] (https://github.com/photopea/UTIF.js/issues/102#issuecomment-1106622739)

**Comment by photopea** [comment]
If you need to store compressed images, I recommend you to use the PNG format instead of TIFF. The PNG is a better format, and it is supported pretty much everywhere.

You can use our library https://github.com/photopea/UPNG.js

**Comment by jardicc** [comment]
We already had this discussion before. TIFF can support all features as PSD due to Adobe's proprietary section + much more. You can even replace PSD with TIFF and produce smaller files due to better bitmap compressions. PNG is just a flat image  ¯\_(ツ)_/¯

If you would use PNGs to recreate layers functionality then you would be reinventing a new image format because you would need some "glue".

If I remember correctly to support this in UTIF... you would need to write about 5 lines of code?

**Comment by photopea** [comment]
"TIFF can support all features as PSD due to Adobe's proprietary section" you are suggesting to make files, which have the "tiff" extension, but are openable only in Adobe Photoshop.

I really dont recommend to do so. You should use the PSD, if you want layers. PSD files can be opened in more programs, even in some free programs.

**Comment by varun-raghavendra** [comment]
@jardicc, could you recall what you did in 2021 for your project? Any code reference would be helpful.

**Comment by jo-chemla** [comment]
Hi there, also interested in adding support for compression to UTIF.js during encoding. Our use case is exporting geotiff from the browser, which has to be tiff. 

LZW, deflate, would be useful. 

Seems according to [this comment](https://github.com/photopea/UTIF.js/issues/132#issuecomment-2621653246) that deflate is already used if pako library is available from window scope. Are you planning to adding more and packaging modules/libraries differently? Thanks!

**Comment by photopea** [comment]
Yes, deflate is already available. LZW is not better than deflate, it is almost always worse.

I think that JPG inside TIFF makes no sense. It used to make sense 30 years ago, when the TIFF format was 8 years old. Today, you should simply use JPG files instead.

**Comment by johnthad** [comment]
> Yes, deflate is already available. LZW is not better than deflate, it is almost always worse.
> 
> I think that JPG inside TIFF makes no sense. It used to make sense 30 years ago, when the TIFF format was 8 years old. Today, you should simply use JPG files instead.

Agreed. Especially as there are two competing "standards": The JPEG-in-TIFF described in the TIFF6 spec (June 3, 1992), and the one described in the "[DRAFT TIFF Technical Note #2](https://download.osgeo.org/libtiff/doc/TIFFTechNote2.html)" (March 17, 1995). LIBTIFF 4.7.0 calls the former compression scheme "Old-style JPEG" and the newer just "JPEG". macOS will not open the older, but it will open the newer one. Moreover, I have seen "Old-style JPEG" implemented two ways. All three of these formats are a pain to deal with.

**Comment by photopea** [comment]
I am confused. How exactly can I help you? What exactly geotiff.js can do that UTIF.js can not do?

**Comment by jo-chemla** [comment]
I was just mentioning geotiffjs because its tiff writer features rely partially on utifjs. So if utifjs can add support for some compression like LZW or JPG during encoding, then these could be ported to geotiffjs as well at some point.

**Comment by photopea** [comment]
@jo-chemla I did not see any mention of UTIF.js in geotiff.js, so I don't think that it relies on UTIF.js

**Comment by photopea** [comment]
@jardicc Right now, Photopea can open every file Photoshop can open, including all the "aspcts" of these files.

BTW. if you export layered TIFF files from Photoshop, they are always bigger than a PSD file would be.

**Comment by jo-chemla** [comment]
It's just for the GeotiffWriter here https://github.com/geotiffjs/geotiff.js/blob/master/src%2Fgeotiffwriter.js#L2
